packages/core/src/storage/compare.ts

```diff
@@ -91,13 +96,38 @@ export class StorageLayoutComparator {
     }
   }
 
+  // Verify the original type and updated real type, not the reported one, are actually compatible
+  getLayoutChange(original: StorageField, updated: StorageField): boolean {
+    const validPair = ['uint8', 'bool'];
+    const knownCompatibleTypes =
+      validPair.includes(original.type.item.label) && validPair.includes(updated.type.item.label);
+    // This is so it doesnt throw an error when reading slot and offset
+    const anyOriginal: any = original;
+    const anyUpdated: any = updated;
+    if (original.type.item.label == updated.type.item.label || knownCompatibleTypes) {
+      return false;
+    } else if (anyOriginal.slot && anyOriginal.offset && anyUpdated.slot && anyUpdated.offset) {
+      const originalValues = JSON.stringify({
+        slot: anyOriginal.slot,
+        offset: anyOriginal.offset,
+        numberOfBytes: original.type.item.numberOfBytes,
+      });
+      const updatedValues = JSON.stringify({
+        slot: anyUpdated.slot,
+        offset: anyUpdated.offset,
+        numberOfBytes: updated.type.item.numberOfBytes,
+      });
+      return !(originalValues === updatedValues);
```

Comment:
Why are we comparing JSON strings? :scream: 

We should just compare the values directly.

Reply:
I thought this was going to be easier to read, but I'll update it.